tskv/src/database.rs

```diff
@@ -147,33 +143,49 @@ impl Database {
         // (series id, schema id) -> RowGroup
         let mut map = HashMap::new();
         for point in points {
-            let mut info =
-                SeriesInfo::from_flatbuffers(&point).context(error::InvalidModelSnafu)?;
+            let (mut info, row) = {
+                (SeriesInfo::from_flatbuffers(&point).context(error::InvalidModelSnafu)?,
+                 RowData::from(point))
+            };
 
             let sid = self.build_index_and_check_type(&mut info)?;
 
-            let row = RowData::from(point);
-            let mut all_fileds = BTreeMap::new();
-            let fields = info.field_infos();
-            for (i, f) in row.fields.into_iter().enumerate() {
-                all_fileds.insert(fields[i].field_id(), f);
-            }
+            let all_fileds = {
+                let mut all_fileds = BTreeMap::new();
+                let fields = info.field_infos();
+                for (i, f) in row.fields.into_iter().enumerate() {
+                    all_fileds.insert(fields[i].field_id(), f);
+                }
 
-            let fields = info.field_fill();
-            for i in 0..fields.len() {
-                all_fileds.insert(fields[i].field_id(), None);
-            }
+                let fields = info.field_fill();
+                for i in 0..fields.len() {
+                    all_fileds.insert(fields[i].field_id(), None);
+                }
+                all_fileds
+            };
 
             let ts = row.ts;
             let schema_id = info.get_schema_id();
-            let mut schema = Vec::with_capacity(all_fileds.len());
-            let mut all_row = Vec::with_capacity(all_fileds.len());
-            for (k, v) in all_fileds {
-                let (fid, _) = split_id(k);
+            let (schema, all_row, schema_size, row_size) = {
+                let mut schema = Vec::with_capacity(all_fileds.len());
+                let mut all_row: Vec<Option<FieldVal>> = Vec::with_capacity(all_fileds.len());
+                let schema_size = schema.capacity() * size_of::<u32>();
+                let mut row_size = all_row.capacity() * size_of::<Option<FieldVal>>();
+
+                for (k, field_val) in all_fileds {
+                    let (fid, _) = split_id(k);
+                    schema.push(fid);
+
+                    row_size += size_of_val(&field_val);
+                    if let Some(FieldVal::Bytes(v)) = &field_val {
+                        row_size += v.capacity();
+                    }
```

Comment:
sizeof Option<FieldVal>  include sizeof FieldVal, and FieldVal::size() caculate (sizeof FieldVal + sizeof Bypass tes)